source/PlayerInfo.cpp

```diff
@@ -1796,7 +1817,7 @@ void PlayerInfo::UpdateCargoCapacities()
 	int bunks = 0;
 	flagship = FlagshipPtr();
 	for(const shared_ptr<Ship> &ship : ships)
-		if(ship->GetSystem() == system && !ship->IsParked() && !ship->IsDisabled())
+		if(ship->GetPlanet() == planet && !ship->IsParked() && !ship->IsDisabled())
```

Comment:
I guess we can remove this check:
```suggestion
		if(ship->GetPlanet() == planet && !ship->IsParked())
```

Reply:
seems like it